Repository Detail Page

bradley-holt opened the pull request:
This pull request creates a per-repo metrics detail page. In its current iteration, this page is fairly useless as it simply provides a redundant view of the same data from the overview page. However, this pull request sets the groundwork for some upcoming features:
1. More detailed information about deployments of each specific repository
2. An HTML widget that can be embedded in tutorials, blog posts, etc.
3. An image widget that can be embedded in READMEs (similar to Travis CI build status images)

Some notes about this new feature:
- An MD5 hash is created of the repository URL and stored in each tracking event.
- A script is run on deployment to add hashes to previously tracked events (this can eventually be removed).
- A `by_repo_hash` map/reduce view indexes tracking events based on the repository URL hash.
- The URL hash is used in the page URL for the detail view, which is linked to from each repository instance on the overview page.

bradley-holt committed:
Collect repository URL hash along with the repository URL

bradley-holt committed:
Create a view of repos by hash

bradley-holt committed:
Create and run a task to add repo URL hashes to events without them

bradley-holt committed:
Display metrics for a specific repo

bradley-holt committed:
Provide drill down link to repo-specific stats

bradley-holt committed:
Add missing semicolon.

bradley-holt committed:
Merge branch 'master' into per-repo-metrics

bradley-holt committed:
Change wording

bradley-holt committed:
Add breadcrumb navigation

jsloyer commented:
Left a couple comments.  How do you handle when an event is already stored that does not have a hash?

bradley-holt committed:
Add URL hash to overview stats page

bradley-holt committed:
Add URL hash to detail stats page

bradley-holt commented:
@jsloyer In response to your comments:

I removed `aria-hidden="true"` as it should have been removed when the glyphicon was removed.

Sorting by count (reverse or otherwise) would require chained map/reduce, as we would need to sort on the results of a reduce. Well not ideal, this approach of sorting in-memory avoids the complexities associated with chained map/reduce.

As for events previously stored from before this code was introduced:

> A script is run on deployment to add hashes to previously tracked events (this can eventually be removed).

I prefer to use hashes to avoid unexpected issues around URL encoding, URL length, etc.

I've added a hash to both the overview and detail stats pages.

jsloyer commented:
+1.

@bradley-holt you have merge issues, correct and then Ill approve.

bradley-holt committed:
Merge branch 'master' into per-repo-metrics

bradley-holt commented:
@jsloyer Merge issue has been resolved.
